src/forms/RadioButtonGroup.jsx

```diff
@@ -1,73 +1,68 @@
 import React, { PureComponent } from 'react';
 import PropTypes from 'prop-types';
 
-import Flex from '../layout/Flex';
-import FlexItem from '../layout/FlexItem';
-
 /**
  * @class RadioButtonGroup
  */
 export default class RadioButtonGroup extends PureComponent {
 	constructor(props) {
 		super(props);
-		this.state = {
-			selectedValue: props.selectedValue,
+
+		const initialState = {
+			value: props.selectedValue,
 		};
-		this.handleChange = this.handleChange.bind(this);
+
+		React.Children.forEach(props.children, radio => {
+			initialState[radio.props.value] = radio.props.value === props.selectedValue;
+		});
+
+		this.state = initialState;
+		this.onChange = this.onChange.bind(this);
+		this.isSelected = this.isSelected.bind(this);
 	}
 
 	componentWillReceiveProps(nextProps) {
 		if ('selectedValue' in nextProps) {
-			this.setState({
-				selectedValue: nextProps.selectedValue,
+			const newState = {};
+
+			Object.keys(this.state).forEach(stateKey => {
+				if (stateKey === 'value') {
+					newState.value = nextProps.selectedValue;
+				} else {
+					newState[stateKey] = stateKey === nextProps.selectedValue;
+				}
 			});
+
+			this.setState(newState);
```

Comment:
I don't particularly like this approach. Would have liked to use hooks. Also, I think that I should move all this logic to `onChange` and that having a `componentWillReceiveProps` method is completely unnecessary, but this is how the old implementation was doing it.

Reply:
yeah this component should probably not be used at all given how re-rendering will reset its internal state. We probably need a backlog ticket to rewrite it correctly